physical/postgresql.go

```diff
@@ -0,0 +1,268 @@
+package physical
+
+import (
+	"database/sql"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"sync"
+	"time"
+)
+
+// PostGresBackend is a physical backend that stores data on postgres database
+type PostGresBackend struct {
+	Url string
+
+	l sync.Mutex
+}
+
+var (
+	Trace   *log.Logger
+	Info    *log.Logger
+	Warning *log.Logger
+	Error   *log.Logger
+)
+
+// newFileBackend constructs a Filebackend using the given directory
+func newPostGresBackend(conf map[string]string) (Backend, error) {
+
+	Trace = log.New(ioutil.Discard, "Trace: ", log.Ldate|log.Ltime|log.Lshortfile)
+	Info = log.New(os.Stdout, "INFO: ", log.Ldate|log.Ltime|log.Lshortfile)
+	Warning = log.New(os.Stdout, "WARNING: ", log.Ldate|log.Ltime|log.Lshortfile)
+	Error = log.New(os.Stdout, "ERROR: ", log.Ldate|log.Ltime|log.Lshortfile)
+
+	Trace.Println("> newPostGresBackend")
+	url, ok := conf["url"]
+	if !ok {
+		return nil, fmt.Errorf("'url' must be set")
+	}
+
+	db, err := sql.Open("postgres", url)
+	if err != nil {
+		Error.Fatal(err)
+	}
+	defer db.Close()
+	sqlStmt := "CREATE EXTENSION if not exists \"uuid-ossp\";"
+	_, err = db.Exec(sqlStmt)
+	if err != nil {
+		Error.Printf("%q: %s\n", err, sqlStmt)
+		Error.Println("< newPostGresBackend(err)")
+		Error.Panic(err)
+	}
+
+	sqlStmt = `
+CREATE TABLE if not exists vault (
+  id UUID PRIMARY KEY NOT NULL DEFAULT uuid_generate_v4(),
+  key TEXT not null,
```

Comment:
I noticed the MySQL backend [uses the `key` as the primary key](https://github.com/hashicorp/vault/blob/master/physical/mysql.go#L84). Is it worth doing the same here, or at least adding a uniqueness constraint on the `key` field? [This thread](http://www.postgresql.org/message-id/1185022160.688503.236330@n2g2000hse.googlegroups.com) seems to imply that the cost of maintaining two indexes may not be worthwhile, especially if the id index is never used.